### PR TITLE
Add traceToStdout and traceToStderr, deprecating traceToIO

### LINKED ISSUE #408: `traceToIO` with `stderr`?
Intended use of `Trace` is for logging but it's main interpreter `traceToIO` is hardcoded to output to `stdout` while more common place to put error/diagnostics/logs for unix programs is `stderr`. 

Not sure what's the right action here but, imo, in ideal world `traceToIO` should output to `stderr` by default.

### PR DESCRIPTION
**Pull request by tek**
fixes #408

### CODE REVIEW

File: src/Polysemy/Trace.hs [hunk 1]
```diff
@@ -29,14 +33,45 @@ data Trace m a where
...
+{-# INLINE traceToHandle #-}
+
+
+------------------------------------------------------------------------------
+-- | Run a 'Trace' effect by printing the messages to stdout.
+--
+-- @since TODO
+traceToStdout :: Member (Embed IO) r => Sem (Trace ': r) a -> Sem r a
+traceToStdout =
+  traceToHandle stdout
```
Comment (KingoftheHomeless) [inline src/Polysemy/Trace.hs:52]: Any reason these aren't one-liners?

Comment (tek) [reply src/Polysemy/Trace.hs:52]: you mean as in unfolding `traceToHandle`?

Comment (KingoftheHomeless) [reply src/Polysemy/Trace.hs:52]: No, I mean
```
traceToStdout =traceToHandle stdout
```

Comment (tek) [reply src/Polysemy/Trace.hs:52]: oh! no, I just always put a newline after the `=`. gonna fix

**Review by KingoftheHomeless — APPROVED** [review]

**Pull request merged**